crypto/threads_none.c

```diff
@@ -25,19 +25,22 @@ CRYPTO_RWLOCK *CRYPTO_THREAD_lock_new(void)
 
 int CRYPTO_THREAD_read_lock(CRYPTO_RWLOCK *lock)
 {
-    OPENSSL_assert(*(unsigned int *)lock == 1);
+    if (!ossl_assert(*(unsigned int *)lock == 1))
+        return 0;
```

Comment:
> We have detected an error so we should just report it.

"Fun" question in the context. How many invocations to this function check for return value? Answer: 0. As for being doomed or not. It's also about assumptions. You say "it's possible that it's use-after-free". Yeah, but it's also possible that it's genuine corruption. We can't tell. But if we can't, do we have to assume that that it's least harmful of all possibilities? But in either case, the comment was meant to serve as an illustration to my argument, i.e. that there *are* situations when it's not inappropriate to perform controlled crash even in library code.